src/transformers/models/marian/tokenization_marian.py

```diff
@@ -170,12 +170,19 @@ def _tokenize(self, text: str) -> List[str]:
         return code + pieces
 
     def _convert_id_to_token(self, index: int) -> str:
-        """Converts an index (integer) in a token (str) using the encoder."""
+        """Converts an index (integer) in a token (str) using the decoder."""
         return self.decoder.get(index, self.unk_token)
 
     def convert_tokens_to_string(self, tokens: List[str]) -> str:
-        """Uses target language sentencepiece model"""
-        return self.spm_target.DecodePieces(tokens)
+        """
+        _decode_use_source_tokenizer is True if use_source_tokenizer is passed to decode/ batch_decode and is True.
+        use_source_tokenizer determines if the source tokenizer should be used to decode sequences (only applicable in
+        sequence-to-sequence problems).
```

Comment:
I think we don't need to document it here, I think what Sylvain meant is, 

override the `decode` and `batch_decode` methods in the `MarianTokenizer` and document the `use_source_tokenizer` argument there. Users will never see the `convert_tokens_to_string` when decoding as it's mostly used internally.

But let's wait for @sgugger  to confirm :)

The rest of the changes look good to me. Thanks a lot!

Reply:
Yes, this is indeed what I meant.

Reply:
No problem! I meant to send a comment after my changes and ask if you prefer these changes or you still want me to copy over decode and batch_decode to MarianTokenizer. I didn't take into account that users won't use convert_tokens_to_string directly and it's better not to hide the logic somewhere else as Patil suggested.
I will amend and push again.